pytorch_lightning/trainer/trainer.py

```diff
@@ -2661,6 +2662,53 @@ def weights_summary(self, val: Optional[str]) -> None:
     Other
     """
 
+    @property
+    def estimated_num_optimization_steps(self) -> Union[int, float]:
+        r"""
+        Estimated number of optimization steps for the complete training inferred from DataLoaders, gradient
+        accumulation factor and distributed setup.
+
+        Examples::
+
+            def configure_optimizers(self):
+                optimizer = ...
+                scheduler = torch.optim.lr_scheduler.OneCycleLR(
+                    optimizer, max_lr=1e-3, total_steps=self.trainer.estimated_num_optimization_steps
+                )
+                return [optimizer], [scheduler]
+
+        """
+        accumulation_scheduler = self.accumulation_scheduler
+
+        if accumulation_scheduler.epochs != [0]:
+            raise MisconfigurationException(
+                "Estimated optimization steps cannot be computed with different"
+                " `accumulate_grad_batches` at different epochs."
+            )
+
+        # infinite training
+        if self.max_epochs == -1 and self.max_steps == -1:
+            return float("inf")
+
+        if self.train_dataloader is None:
+            rank_zero_info("Loading `train_dataloader` to estimate number of optimization steps.")
+            self.reset_train_dataloader()
+
+        total_batches = self.num_training_batches
+
+        # iterable dataset
+        if total_batches == float("inf"):
+            return self.max_steps
+
+        self.accumulate_grad_batches = accumulation_scheduler.get_accumulate_grad_batches(self.current_epoch)
+        effective_batch_size = self.accumulate_grad_batches
+        max_estimated_steps = math.ceil(total_batches / effective_batch_size) * (
+            self.max_epochs if self.max_epochs != -1 else 1
+        )
```

Comment:
```suggestion
        max_estimated_steps = math.ceil(total_batches / effective_batch_size) * max(self.max_epochs, 1)
```
or is there a case with `self.max_epochs` is 0?

Reply:
I don't think so.. but just kept to avoid zero-division error in any case.

Reply:
nope..